dev/com.ibm.ws.artifact.zip/src/com/ibm/ws/artifact/zip/internal/JavaArchive.java

```diff
@@ -0,0 +1,575 @@
+	private static final int ZIP64_LOCATOR_EOCD_OFFSET_OFFSET = 8;   // Position in the locator of the offset of the Zip64-End-of-Central-Direcory-Record
+	
+	private static final int LOCAL_FILE_HEADER_SIGNATURE = 0x504b0304;
+	private static final int BLOCK_SIZE = 256;
+	private static final int FORWARD_BLOCK_SIZE = 8192;
+	
+	// Maximum amount of data (usually a script) that is allowed by this implementation to be 
+	// attached to the beginning of the archive. This value may be increased, but it will 
+	// increase the time it takes to fail when the zip is corrupted.
+	private static final int MAX_PREFIX_ALLOWED = 1024*1024*5;       
+	
+	private byte[] _endOfCentralDirectoryRecord; // Only the minimum size is stored.  Does not include the comment 
+	private int    _eocdRecordSize;              // Actual size of End-of-Central-Directory-Record - including comment
+	private byte[] _zip64EoCDRLocator;           // 20 bytes;
+	private long   _zip64EoCDRLocatorOffset;     // Offset (from end of file) of the Zip64 Locator
+	private byte[] _zip64EoCDR;                  // Only the minimum size is stored.  Does not include any extra fields.  
+	private long   _zip64EoCDRecSize;            // Size of Zip64-End-of-Central-Directory-Record
+    private String _archiveFileName;
+	private long   _fileLength;                  // Length of the archive file
+	
+    JavaArchive(String archiveFileName) {
+    	_archiveFileName = archiveFileName;
+    }
+    
+    public boolean isValid() {
+    	
+    	try (  RandomAccessFile file = new RandomAccessFile(_archiveFileName, "r") ) {
+
+    		_fileLength = file.length();
+    		_endOfCentralDirectoryRecord = getEndOfCentralDirectoryRecord(file);
+    		
+    		if ( isZip64() ) {
+    		   _zip64EoCDRLocator = getZip64EoCDLocator(file);
+    		   _zip64EoCDRLocatorOffset = _fileLength - _eocdRecordSize - ZIP64_LOCATOR_SIZE;
+    		   _zip64EoCDR = getZip64EndOfCentralDirectoryRecord(file);
+    		   _zip64EoCDRecSize = getZip64EoCDRecordSize( _zip64EoCDR, 0 );
+    		}
+    		
+    		int archiveOffset = (int)getActualArchiveOffset();
+
+    		if (archiveStartsWithSignature(file, archiveOffset)) {
+    			return true;
+    		} 
+
+    	} catch (Exception e) {
+    		//e.printStackTrace();
+    	}
+
+    	return false;
+    }
+    
+    /**
+     * Find the End-of-Central-Director-Record which is always at the end of the file.
+     * It is usually 22 bytes but may contain a comment up to 64K in size.
+     * Reads from the end of the file looking backwards for the EoCDR signature.
+     * 
+     * The implementation reads the file in BLOCK_SIZE blocks from the end of the file
+     * and searches each block.  Note that there is some overlap of the successive 
+     * blocks; EOCDR_MIN_SIZE bytes of overlap to be precise.  This is because the 
+     * record might start at the end of one block and finish in the next.
+     * 
+     * @param file
+     * @throws IOException
+     */
```

Comment:
Either add documentation on the @param and @throws lines, or omit them.

Reply:
done